detox/android/detox/src/main/java/com/wix/detox/DetoxInstrumentsManager.java

```diff
@@ -0,0 +1,141 @@
+package com.wix.detox;
+
+import android.content.Context;
+import android.util.Log;
+
+import java.io.File;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Method;
+
+public class DetoxInstrumentsManager {
+    private static Constructor constructorDtxProfilingConfiguration;
+    private static Method methodGetInstanceOfProfiler;
+    private static Method methodStartRecording;
+    private static Method methodStopRecording;
+    private static Method methodEventBeginInterval;
+    private static Method methodEventEndInterval;
+    private static Method methodEventMark;
+    private static final boolean hasProfiler;
+
+    static {
+        try {
+            final String basePackageName = "com.wix.detoxprofiler";
+            final Class<?> profilerClass = Class.forName(basePackageName + ".DTXProfiler");
+            final Class<?> configurationClass = Class.forName(basePackageName + ".DTXProfilingConfiguration");
+
+            constructorDtxProfilingConfiguration = configurationClass.getConstructor(
+                    boolean.class,//recordPerformance
+                    long.class,//samplingIntervalMillis
+                    File.class,//recordingFile
+                    boolean.class//recordReactNativeTimersAsEvents
+            );
+            methodGetInstanceOfProfiler = profilerClass.getDeclaredMethod("getInstance", Context.class);
+            methodStartRecording = profilerClass.getDeclaredMethod("startProfiling", Context.class, configurationClass);
+            methodStopRecording = profilerClass.getDeclaredMethod("stopProfiling");
+            methodEventBeginInterval = profilerClass.getDeclaredMethod("eventBeginInterval", String.class, String.class, String.class, String.class, String.class);
+            methodEventEndInterval = profilerClass.getDeclaredMethod("eventEndInterval", String.class, String.class, String.class);
+            methodEventMark = profilerClass.getDeclaredMethod("eventMark", String.class, String.class, String.class, String.class, String.class);
+        } catch (Exception e) {
+            Log.i("DetoxInstrumentsManager", "Instruments not found", e);
+
+            constructorDtxProfilingConfiguration = null;
+            methodGetInstanceOfProfiler = null;
+            methodStartRecording = null;
+            methodStopRecording = null;
+            methodEventBeginInterval = null;
+            methodEventEndInterval = null;
+            methodEventMark = null;
+        }
+        hasProfiler = methodGetInstanceOfProfiler != null;
+    }
+
+    private final Context context;
+    private Object dtxProfilerInstance;
+
+    DetoxInstrumentsManager(Context context) {
```

Comment:
Please specify which context type exactly you're expecting here (we have too many in instrumentation testing...)

Reply:
This is still applicable